### DISCUSSION TITLE
Increment/decrement loses precision with decimal columns

### DISCUSSION THREAD

**Discussion by drishit96** [prisma/prisma#18518] [Q&A] [answered]
I am trying to increment a column of type `Decimal @db.Decimal(22, 4)`. It works for numbers which are supported by the `number` primitive of javascript. However, when I increment by a bigger number, it starts to lose precision.
This may be a known limitation as the [docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#atomic-number-operations) mention only `Float` and `Int`.
I tried incrementing with raw query as well but the result is the same. I think prisma is converting the value to number even though it's a decimal.
I am using prisma v4.10.1
Any idea how I can increment on a decimal type without precision loss?

**Comment by nurul3101** [comment, marked as answer]
Hello @drishit96 👋 

Prisma uses [Decimal.js](https://github.com/MikeMcl/decimal.js/) for the Decimal support.
Can you check if you get the same precision loss if you try to add two decimals with the Decimal.js library?

**Reply by drishit96** [reply]
The existing value of the column was 0 and I was incrementing it by the bigger number. So, I don't think decimal.js is causing any issue. Also, I did pass the value to the Number constructor and saw that the value was the same that was being saved to the db.

**Reply by nurul3101** [reply]
Okay, I believe as the docs mention `Int` and `Float` only, then it might be expected. Do you mind opening a [GitHub Issue](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) for it?
